### PR TITLE
[8.x] Fixed the wrong original values of incrementing with extra.

### PR DESCRIPTION
**Pull request by limingxinleo**

### CODE REVIEW

File: tests/Database/DatabaseEloquentModelTest.php [hunk 1]
```diff
@@ -1691,7 +1691,7 @@ public function testIncrementOnExistingModelCallsQueryAndSetsAttribute()
         $model->publicIncrement('foo', 1, ['category' => 1]);
         $this->assertEquals(4, $model->foo);
         $this->assertEquals(1, $model->category);
-        $this->assertTrue($model->isDirty('category'));
```
Comment (limingxinleo) [inline tests/Database/DatabaseEloquentModelTest.php:1694]: The test case...

I think it is not correct, when id changed the extra data, it shoud be not dirty.

**Comment by driesvints** [comment]
There's no explanation at all what this fixes so this PR is highly likely to be closed.

**Comment by limingxinleo** [comment]
$m->increment($value,1,["data" => 1]);
$m->save();

data will be saved twice.

**Comment by taylorotwell** [comment]
The fact that an existing test had to be changed to its opposite assertion makes me worry this is a bigger breaking change.

**Pull request closed without merging**